### DISCUSSION TITLE
How to improve my code?

### DISCUSSION THREAD

**Discussion by carl336** [micropython/micropython#14038] [RP2040 / Pico]
Hi, I am trying to turn on the Backlights by using Raspberry Pi Pico with Micropython.

(MicroPython v1.22.2 on 2024-02-22; Raspberry Pi Pico with RP2040)
(the BL that I'm using has 17 LED Drivers, each LED Driver controls 48 LEDs. In total, there are 816 LEDs.)

Here is the program code that I wrote:
https://paste.fastmirror.net/?950670a4c6475889#CsY93vo2wfg7W7Ko4H7WrpLKvZNu8yKHUQeUyT91aUJe 

this is how I set up the LEDs to light up, you can watch it by this URL:
https://www.veed.io/view/8e4a00a4-c7b8-4fbc-95c4-5ebcb507813e?panel=share

Also, I just uploaded some files to a website, you may download them for free to check it.

https://file.io/AezvCEc9rDUd

Those files are:

a circuit diagram that I draw,

a LED circuit diagram inside the backlight board,

a specification for LEDs that are used,

PWM LED Driver with Internal Current Settings,

A drawing of the specific position of each LED that I draw.

**Comment by GitHubsSilverBullet** [comment]
Don't get me wrong, but that code can't be improved.
Move it from pastebin to wastebin! Throw it away entirely. 

Then learn the Python basics and try again.
Hint:
If it barely does anything useful, then 10000 lines of code are a bit to much.
As a starter, learn about array and how to properly fill them with ever repeating values.

**Comment by sosi-deadeye** [comment]
Don't repeat yourself was completely ignored. All LED-Patterns could be created dynamically with one function or could be saved static on the flash storage.

A complete rewrite should be done. But before you're doing this, learn Python.

**Comment by karfas** [comment]
@carl336: You already got some answers in https://github.com/orgs/micropython/discussions/13726.

Most likely, you will not get different answers when asking the same thing over and over.
